ddtrace/internal/processor/trace.py

```diff
@@ -127,8 +127,14 @@ def on_span_finish(self, span):
             if trace.num_finished == len(trace.spans) or (
                 self._partial_flush_enabled and trace.num_finished >= self._partial_flush_min_spans
             ):
-                finished = [s for s in trace.spans if s.finished]
-                trace.spans = [s for s in trace.spans if not s.finished]
+                trace = trace.spans
+                finished = []
+                trace.spans = []
+                for s in trace:
```

Comment:
we probably also only need to do this if `trace.num_finished < len(trace.spans)`, otherwise we can just do `finished = trace.spans; trace.spans = []` and skip the loop entirely